Modules/Devices/src/Devices.Domain/Entities/Identities/CommunicationLanguage.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Globalization;
+using Backbone.BuildingBlocks.Domain.Errors;
+using CSharpFunctionalExtensions;
+
+namespace Backbone.Modules.Devices.Domain.Entities.Identities;
+
+public record CommunicationLanguage
+{
+    public static readonly CommunicationLanguage DEFAULT_LANGUAGE = new("en");
+
+    public string Value { get; }
+    public const int LENGTH = 2;
+
+    private CommunicationLanguage(string value)
+    {
+        Value = value;
+    }
+
+    public static Result<CommunicationLanguage, DomainError> Create(string value)
+    {
+        var validationResult = Validate(value);
+        if (validationResult != null)
+            return Result.Failure<CommunicationLanguage, DomainError>(validationResult);
+        return Result.Success<CommunicationLanguage, DomainError>(new CommunicationLanguage(value));
+    }
+
+    public static DomainError? Validate(string value)
+    {
+        if (value.Length != LENGTH)
+            return DomainErrors.InvalidDeviceCommunicationLanguage($"Device Communication Language length must be {LENGTH}");
```

Comment:
I don't think this check makes sense. If it is longer than 2 characters, we will never find it in the list of allowed values, so it will be considered invalid nonetheless